adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -251,6 +256,14 @@ public static AuthenticationResult processUIResponseParams(Map<String, String> r
 
         } else if (response.containsKey(AuthenticationConstants.OAuth2.CODE)) {
             result = new AuthenticationResult(response.get(AuthenticationConstants.OAuth2.CODE));
+            final String cloudInstanceName = response.get(AuthenticationConstants.OAuth2.CLOUD_INSTANCE_NAME);
+            if (!StringExtensions.isNullOrBlank(cloudInstanceName)) {
+                final StringBuilder cloudInstanceUrl = new StringBuilder("https://login.")
```

Comment:
Use the passed in authority url, break it down to segments and only replace the cloud instance.